guides/coco-dev.md

```diff
@@ -0,0 +1,248 @@
+and confidential hardware.
+
+CoCo has a modular attestation interface and there are a few options for attestation.
+CoCo provides a generic Key Broker Service (KBS) that the rest of this guide will be focused on.
+The SEV runtime class uses `simple-kbs`, which is described in the [SEV guide](../guides/sev.md).
+
+### Select Runtime Class
+
+To use CoCo with confidential hardware, first switch to the appropriate runtime class.
+TDX has one runtime class, `kata-qemu-tdx`. 
+
+For SEV(-ES) use the `kata-qemu-sev` runtime class and follow the [SEV guide](../guides/sev.md). 
+
+For SNP, use the `kata-qemu-snp` runtime class and follow the SNP guide.
+
+For `enclave-cc` follow the [enclave-cc guide](../guides/enclave-cc.md).
+
+### Deploy and Configure tenant-side CoCo Key Broker System cluster
+
+The following describes how to run and provision the generic KBS.
+The KBS should be run in a trusted environment. The KBS is not just one service,
+but a combination of several.
+
+A tenant-side CoCo Key Broker System cluster includes:
+- Key Broker Service (KBS): Brokering service for confidential resources.
+- Attestation Service (AS): Verifier for remote attestation.
+- Reference Value Provider Service (RVPS): Provides reference values for AS.
+- CoCo Keyprovider: Component to encrypt the images following ocicrypt spec.
+
+To quick start the KBS cluster, a `docker compose` yaml is provided to launch.
+
+```shell
+# Clone KBS git repository
+git clone https://github.com/confidential-containers/trustee.git
+cd trustee/kbs
+export KBS_DIR_PATH=$(pwd)
+
+# Generate a user auth key pair
+openssl genpkey -algorithm ed25519 > config/private.key
+openssl pkey -in config/private.key -pubout -out config/public.pub
+
+cd ..
+
+# Start KBS cluster
+docker compose up -d
+```
+
+If configuration of KBS cluster is required, edit the following config files and restart the KBS cluster with `docker compose`:
+
+- `$KBS_DIR_PATH/config/kbs-config.toml`: configuration for Key Broker Service.
+- `$KBS_DIR_PATH/config/as-config.json`: configuration for Attestation Service.
+- `$KBS_DIR_PATH/config/sgx_default_qcnl.conf`: configuration for Intel TDX/SGX verification.
+
+When KBS cluster is running, you can modify the policy file used by AS policy engine ([OPA](https://www.openpolicyagent.org/)) at any time:
+
+- `$KBS_DIR_PATH/data/attestation-service/opa/default.rego`: Policy file for evidence verification of AS, refer to [AS Policy Engine](https://github.com/confidential-containers/attestation-service#policy-engine) for more infomation.
+
+### Encrypting an Image
+
+[skopeo](https://github.com/containers/skopeo) is required to encrypt the container image.
+Follow the [instructions](https://github.com/containers/skopeo/blob/main/install.md) to install `skopeo`.
+Make sure to follow the instructions to build skopeo from source, otherwise there will be errors regarding version
+incompatibility between ocicrypt and skopeo.
```

Comment:
What do we now about the incompatible versions? Readers might want to check if their distro already provides new enough version to avoid building.

Reply:
What I've found online is that we need at least skopeo 1.16, because the one Ubuntu 22.04 downloads skopeo 1.4.1.

Reply:
What I've found online is that we need at least skopeo 1.16, because the one Ubuntu 22.04 downloads skopeo 1.4.1.

Reply:
I don't think that is true. skopeo 1.4.1 should be fine but Ubuntu builds it with an ancient ocicrypt version which does not support the keyprovider protocol we depend on. Ubuntu 24.04 LTS is fine already so I don't think we should say "build your own" unconditionally.

Reply:
I've made the update with that information. Thank you for the clarification.

Reply:
thanks, there was a small typo but other than that looks fine